annotation-processor/src/main/kotlin/pl/touk/krush/model/EntityGraphBuilder.kt

```diff
@@ -39,34 +38,39 @@ class EntityGraphBuilder(
         return graphs
     }
 
-    private fun buildEntities(entityList: List<TypeElement>, graphs: EntityGraphs) {
+    private fun buildEntities(entityList: List<TypeElement>) : EntityGraphs {
+        val graphs = EntityGraphs()
         for (entityElt in entityList) {
             val graph = graphs.getOrDefault(entityElt.packageName, EntityGraph())
             graph[entityElt] = EntityDefinition(
                     name = entityElt.simpleName, qualifiedName = entityElt.qualifiedName, table = entityElt.tableName
             )
             graphs[entityElt.packageName] = graph
         }
+        return graphs
     }
 
     private fun AnnotationEnvironment.validate(): ValidationResult {
-        return entities.doValidate { entityElt ->
-            entityTypeValidators.map { validator -> validator.validate(entityElt) }
-        }
+        return entities.doValidate(entityTypeValidators)
     }
 
     private fun EntityGraphs.validate(): ValidationResult {
-        return entities().doValidate { entityDefinition ->
-            entityDefValidators.map { validator -> validator.validate(entityDefinition) }
-        }
+        return entities().doValidate(entityDefValidators)
     }
 
-    private fun <T> Iterable<T>.doValidate(validationExpr: (T) -> List<ValidationResult>): ValidationResult {
+    private fun <T> Iterable<T>.doValidate(validators: List<Validator<T>>): ValidationResult {
         val errors = mutableListOf<ValidationErrorMessage>()
 
         this.forEach { el ->
-            val elValidationResults = validationExpr(el)
-            elValidationResults.filterIsInstance<Error>().forEach { errors.addAll(it.errors) }
+            validators.forEach { validator ->
+                try {
+                    when (val elValidationResult = validator.validate(el)) {
+                        is Error -> errors.addAll(elValidationResult.errors)
+                    }
+                } catch (ex: Exception) {
```

Comment:
\[detekt\] ERROR: \[exceptions/CatchException\] Caught exception is too generic\. Prefer catching specific Exceptions to handle specific error cases\.